### ISSUE TITLE
Crash in BuildUnitAction due to discovering a Pipe was turned into Rubble.

### ISSUE THREAD

**Issue by Xirema** [DeamonHunter/AWBW-Replay-Player#114]
This replay shows an error occurring on Day 14, when P2 (Pink) builds a Medium Tank at the top-center base.

Curiously, another Medium Tank was built on Day 15 at the same base, but the same error does not occur.

```
System.Collections.Generic.KeyNotFoundException: The given key '116' was not present in the dictionary.
   at AWBWApp.Game.Game.Logic.GameMap.RegisterDiscovery(DiscoveryCollection collection)
   at AWBWApp.Game.API.Replay.Actions.BuildUnitAction.PerformAction(ReplayController controller)+MoveNext()
   at AWBWApp.Game.Game.Logic.ReplayController.Update()
```

[Vs Von Bolt.zip](https://github.com/DeamonHunter/AWBW-Replay-Player/files/9822343/Vs.Von.Bolt.zip)

**Comment by DeamonHunter** [comment]
Looks like a crash due to a unit discovering a pipe that turned into rubble.

I should hopefully have a fix out by tomorrow.

**Comment by DeamonHunter** [comment]
Was fixed with 509b0bcf380d7bd694ca7497580d521fa7fd2ca8.
